### PR TITLE
move rate limiter state mutations to functions

### PR DESCRIPTION
**Pull request by ShanniLi**
r @Raynos @kriskowal @rf

### CODE REVIEW

File: node/rate_limiter.js [hunk 1]
```diff
@@ -124,10 +124,35 @@ function removeServiceCounter(serviceName) {
...
+        self.updateServiceLimit(name, rpsLimitForServiceName[name]);
+    }
+};
+
 RateLimiter.prototype.updateServiceLimit =
 function updateServiceLimit(serviceName, limit) {
     var self = this;
-    self.rpsLimitForServiceName[serviceName] = limit;
+
+    if (limit === undefined) {
```
Comment (anson627) [inline node/rate_limiter.js]: when would limit become undefined?

File: node/hyperbahn/service_proxy.js [hunk 2]
```diff
@@ -509,6 +509,18 @@ function destroy() {
...
 };
 
+ServiceDispatchHandler.prototype.enableRateLimiter =
+function enableRateLimiter() {
+    var self = this;
+    self.rateLimiter.rateLimiterEnabled = true;
+};
+
+ServiceDispatchHandler.prototype.disableRateLimiter =
+function enableRateLimiter() {
```
Comment (anson627) [inline node/hyperbahn/service_proxy.js]: typo?

File: node/rate_limiter.js (hunk 1 above)
Comment (anson627) [inline node/rate_limiter.js]: can limit be 0? Is that what we are trying to do?

File: node/hyperbahn/service_proxy.js (hunk 2 above)
Comment (ShanniLi) [inline node/hyperbahn/service_proxy.js]: good catch! fixed.

File: node/rate_limiter.js (hunk 1 above)
Comment (ShanniLi) [inline node/rate_limiter.js]: I don't really have strong preference. Allowing 0 let us disable a service completely. That is a flexibility we may want.

Comment (anson627) [inline node/rate_limiter.js]: cool, just curious

**Comment by anson627** [comment]
lgtm

File: node/rate_limiter.js [hunk 3]
```diff
@@ -124,10 +124,35 @@ function removeServiceCounter(serviceName) {
...
 };
 
+RateLimiter.prototype.updateExemptServices =
+function updateExemptServices(exemptServices) {
+    var self = this;
+    self.exemptServices = exemptServices;
+};
+
+RateLimiter.prototype.updateRpsLimitForServiceName =
+function updateRpsLimitForServiceName(rpsLimitForServiceName) {
```
Comment (Raynos) [inline node/rate_limiter.js]: Let's name this more clearly.

`updateRpsLimitForAllServices`

File: node/rate_limiter.js (hunk 1 above)
Comment (Raynos) [inline node/rate_limiter.js]: it's cleaner to check for `if (limit === null)` and then have `updateServiceLimit('foo', null)`

Comment (kriskowal) [inline node/rate_limiter.js]: undefined is appropriate.

**Comment by kriskowal** [comment]
Looks good. I concur with Jake’s recommendation for the method name.

**Comment by Raynos** [comment]
#legit

**Pull request merged**